components/build-templates/production/kustomization.yaml

```diff
@@ -4,3 +4,4 @@ resources:
   - ../base
   - e2e-quay-push-secret.yaml
   - e2e-registryredhatio-pull-secret.yaml
+  - serviceaccount.yaml
```

Comment:
nipick: these could be called e2e-serviceaccount.yaml, to make it clear that they're not for the build-templates namespace but for the build-templates-e2e namespace

Reply:
(a bigger refactoring is in order, but IIRC Max already filed a story to get rid of the build-templates namespace so that's covered)

Reply:
I agree, I see no point in renaming the SA now

Reply:
Renamed the filename.